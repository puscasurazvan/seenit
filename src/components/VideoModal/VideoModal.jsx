import React from 'react'

import PropTypes from 'prop-types'

import './VideoModal.scss'

const VideoModal = ({ children, toggle, open }) => {
  return (
    open && (
      <div className="video">
        <div className="video__body">
          {children}
          <button className={'video__body__close-button'} onClick={toggle}>
            How to overcome camera shyness &#10006;
          </button>
        </div>
      </div>
    )
  )
}

VideoModal.propTypes = {
  open: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
  children: PropTypes.any,
}

export default VideoModal
