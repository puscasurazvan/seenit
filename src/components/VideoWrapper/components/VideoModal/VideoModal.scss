.video {
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;

  &__body {
    display: flex;
    flex-direction: column;
    margin: auto;
    padding: 20px;
    width: fit-content;
    align-items: center;

    &__close-button {
      border: none;
      font-size: 20px;
      font-weight: bold;
      color: red;
      animation: fade-in 1.5s cubic-bezier(0.25, 0.46, 0.45, 0.94) forwards;
      @keyframes fade-in {
        0% {
          opacity: 0;
        }
        100% {
          opacity: 1;
        }
      }
    }
    &__close-button:hover {
      cursor: pointer;
      transition: 2.5s;
      font-size: 19px;
      color: black;
    }
  }
}
