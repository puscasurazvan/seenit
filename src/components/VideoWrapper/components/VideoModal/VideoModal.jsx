import React, { useEffect, useState } from 'react'
import { createPortal } from 'react-dom'
import './VideoModal.scss'

const Portal = ({ children }) => {
  const root = document.getElementById('videoWrapper')
  const [element] = useState(document.createElement('div'))

  useEffect(() => {
    root.appendChild(element)

    return function cleanup() {
      root.removeChild(element)
    }
  }, [root, element])

  return createPortal(children, element)
}

const VideoModal = ({ children, toggle, open }) => {
  return (
    <Portal>
      {open && (
        <div className="video">
          <div className="video__body">
            {children}
            <button className={'video__body__close-button'} onClick={toggle}>
              How to overcome camera shyness &#10006;
            </button>
          </div>
        </div>
      )}
    </Portal>
  )
}

export default VideoModal
