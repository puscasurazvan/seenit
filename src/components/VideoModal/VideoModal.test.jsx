import React from 'react'
import { mount } from 'enzyme'
import VideoModal from './VideoModal'

const minProps = {
  open: false,
  toggle: jest.fn(),
}

describe('VideoModal', () => {
  it('should render with the right props', () => {
    const wrapper = mount(<VideoModal {...minProps} />)
    expect(wrapper.props().open).toBe(false)
  })

  it('should render a close button when is opened', () => {
    minProps.open = true
    const wrapper = mount(<VideoModal {...minProps} />)
    expect(wrapper.props().open).toBe(true)
    expect(wrapper.find('button').exists()).toBe(true)
  })

  it('should render without exploding', () => {
    const wrapper = mount(<VideoModal {...minProps} />)
    expect(wrapper.length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })
})
