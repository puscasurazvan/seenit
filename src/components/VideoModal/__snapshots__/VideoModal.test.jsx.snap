// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoModal should render without exploding 1`] = `ReactWrapper {}`;
